components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3436,7 +3436,13 @@ where
         }
 
         let mut meta = self.ctx.store_meta.lock().unwrap();
-        if meta.regions[&self.region_id()] != *self.region() {
+        let region_mismatch = match meta.regions.get(&self.region_id()) {
```

Comment:
This change can avoid the panic, but whether it is sufficient needs more careful thoughts. In some cases, panicking is better than letting the process run and do unexpected things.

Reply:
I think this fix is quite rational. If this region cannot be found in `meta.regions` or dose not equal to the one in `meta.regions`, it means that another peer has replaced the current peer. Maybe we should add more comments in there.

Reply:
Added some comments for it.